subsys/logging/log_backend_fs.c

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <logging/log_backend.h>
+#include "log_backend_std.h"
+#include <assert.h>
+#include <fs/fs.h>
+#if CONFIG_LOG_BACKEND_FS_AUTOSTART
+#include <fs/littlefs.h>
+#endif
+
+#define MAX_PATH_LEN 256
+#define MAX_FLASH_WRITE_SIZE 256
+#define LOG_PREFIX_LEN (sizeof(CONFIG_LOG_BACKEND_FS_FILE_PREFIX) - 1)
+
+enum backend_fs_state {
+	BACKEND_FS_OK = 0,
+	BACKEND_FS_CORUPTED,
+	BACKEND_FS_NOT_INITIALIZED
```

Comment:
i would have it as 0 so that `backend_state` can land in zero initialize section.